### PR TITLE
Move mod/receive to src/Module/Diaspora/receive

### PR DESCRIPTION
**Pull request by nupplaphil**
Part of #4090 

- Added routes
- Make Diaspora::decode(Raw) more explicit
- Add new User::getByGuid() method

### CODE REVIEW

**Comment by nupplaphil** [comment]
Found an error on my node, have to check it!

File: src/Module/Diaspora/Receive.php [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+		switch ($type) {
+			case 'public':
+				self::receivePublic();
+				break;
+			case 'users':
+				self::receiveUser($args->get(2));
+				break;
+			default:
+				self::$logger->info('Wrong call.');
+				throw new HTTPException\InternalServerErrorException('wrong call.');
```
Comment (MrPetovan) [inline src/Module/Diaspora/Receive.php]: This should be a BadRequestException, not an InternalServerErrorException.

File: src/Module/Diaspora/Receive.php [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+			self::$logger->info('Diaspora: Decode message in the old format.');
+			$msg = Diaspora::decode($xml, $privKey);
+
+			if ($public && !$msg) {
+				self::$logger->info('Diaspora: Decode message in the new format.');
+				$msg = Diaspora::decodeRaw($xml, $privKey);
+			}
+		}
+
+		self::$logger->info('Diaspora: Post encoded.');
```
Comment (MrPetovan) [inline src/Module/Diaspora/Receive.php]: `decoded`, rather?

File: src/Module/Diaspora/Receive.php [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+	public static function post()
+	{
+		/** @var Configuration $config */
+		$config = self::getClass(Configuration::class);
+
+		$enabled = $config->get('system', 'diaspora_enabled', false);
+		if (!$enabled) {
+			self::$logger->info('Diaspora disabled.');
+			throw new HTTPException\InternalServerErrorException('Diaspora disabled.');
```
Comment (MrPetovan) [inline src/Module/Diaspora/Receive.php]: Ah damn I didn't see this one that should have been "ForbiddenException" instead.

Comment (nupplaphil) [reply src/Module/Diaspora/Receive.php]: Fixed with #7770

**Pull request merged**